sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -73,6 +74,7 @@ class Analyzer(
       ResolveGroupingAnalytics ::
       ResolvePivot ::
       ResolveUpCast ::
+      ResolveAggregateFunctions ::
```

Comment:
depending on rule orders of one batch is not a good idea. We should either add more `if` conditions to only capture needed cases, or separate them into multiple batches.